virtual_rainforest/models/plants/canopy.py

```diff
@@ -7,17 +7,60 @@
 
 from __future__ import annotations
 
+from dataclasses import dataclass
+
 import numpy as np
 from numpy.typing import NDArray
 from xarray import DataArray
 
-from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.data import Config, Data
 from virtual_rainforest.core.exceptions import ConfigurationError, InitialisationError
 from virtual_rainforest.core.logger import LOGGER
 from virtual_rainforest.core.utils import set_layer_roles
 from virtual_rainforest.models.plants.community import PlantCohort, PlantCommunities
 
 
+@dataclass
+class LayerStructure:
+    """Simulation vertical layer structure.
+
+    This data class is a container for five configurable elements used to define the
+    vertical layer structure of the simulation. All are configured via the
+    ``core.layers`` configuration section.
+    """
+
+    canopy_layers: int
+    """The maximum number of canopy layers."""
+    soil_layers: list[float]
+    """A list of the depths of soil layer boundaries."""
+    above_canopy_height_offset: float
+    """The height above the canopy at which reference climate variables are measured."""
+    surface_layer_height: float
```

Comment:
This is a bit of a tricky one. Most climate data provide the 2m data above ground; if it is not measured they interpolate the data or model output to that. So technically, all the input from ERA5 for example is 2m above ground. For the model, we want the conditions above the canopy as an input. In reality, the downscaling will most likely introduce much more uncertainty than the difference between top of canopy and below canopy, we just need to make this very clear. 
Therefore, it might be better to say `The height above the canopy for which reference climate variables are provided.`